### PR TITLE
Disabling 258 C# Symbol Unit Tests.

### PR DESCRIPTION
**Pull request by tannergooding**
This modifies the failing tests form '[Fact]' to '[ClrOnlyFact]'. This disables them on the failing linux build until a fix can be applied.

### CODE REVIEW

**Comment by tannergooding** [comment]
@jaredpar Can you take a look. I am working on creating an issue to track this.

**Comment by jaredpar** [comment]
:+1:

**Comment by tannergooding** [comment]
This is for Issue #3350.

File: src/Compilers/CSharp/Test/Symbol/BadSymbolReference.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Symbol/BadSymbolReference.cs]: Why would this be showing up as a change? Did you change the line-ending sequence?

**Comment by gafter** [comment]
LGTM

File: src/Compilers/CSharp/Test/Symbol/BadSymbolReference.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Symbol/BadSymbolReference.cs]: @gafter I'm guessing it's actually a BOM encoding change.

Comment (tannergooding) [inline src/Compilers/CSharp/Test/Symbol/BadSymbolReference.cs]: @gafter I can confirm, it looks like my script to change out the [Fact] attributes saved the files without the BOM.

**Comment by eriawan** [comment]
Hi Tanner,
Apologize if my question might seem too obvious:
- what would be the difference of `[Fact]` and `[ClrOnlyFact]` attributes?
- Is the `[ClrOnlyFact]` closely related to CoreCLR in dotnetCore?

Thanks in advance.

**Comment by tannergooding** [comment]
@eriawan. I think @jaredpar could provide a better explanation than myself, but I'll take my best stab at it :).

There are essentially two versions of the runtime; the "portable" version (CoreCLR) and the "desktop" version (CLR). The desktop version represents the full .NET Framework and CLR runtime and is only available on windows; while the portable version, represents the cross-platform subset of the .NET Framework and CLR runtime (these would be the **dotnet\corefx** and **dotnet\coreclr** projects, respectively).

`[Fact]` should indicate that the unit test can be run on both the portable and desktop frameworks. While `[ClrOnlyFact]` should indicate that it can only run on the "desktop" framework (Windows only) due to some dependency or a current limit of the CoreCLR runtime.

The tests were passing on Mac and Windows, and were just recently enabled for Linux. However, it seems there was some issue that is currently preventing the tests from running correctly on Linux. As such, they are temporarily disabled for the Mac/Linux platforms and left enabled on Windows (by switching the attribute from `[Fact]` to `[ClrOnlyFact]`) until the underlying issue can be resolved.

**Comment by dpoeschl** [comment]
@eriawan If you're interested, you can see [how it's implemented and which existing tests use it](http://source.roslyn.io/#Roslyn.Test.Utilities/ConditionalFactAttribute.cs,454e830f369bbf75,references) (and which ```ClrOnlyReason``` they provide). It calls into [this helper](http://source.roslyn.io/#Roslyn.Test.Utilities/CLRHelpers.cs,a50e755a2a155466) to decide whether to skip the test.

**Comment by eriawan** [comment]
@tannergooding thanks for your quick reply!
I think I'll try to test on my VMs, since it's quite new for me to have CoreCLR on machines other than Windows.

@dpoeschl thanks! I'm interested.
Does this mean the `ClrOnlyReason` is currently to handle unsupported tests scenarios on Mono? Please CMIIW.

**Comment by jaredpar** [comment]
@eriawan the idea here is that there are a couple of factors which could influence whether or not a test will succeed on a given run:

- Runtime: .NET Desktop, CoreCLR, Mono
- Platform: Windows, Linux, Mac

To support disabling relevant tests on these configurations we have the following attributes:

- [ConditionalFact(typeof(WindowsOnly))]: Only run the test on Windows
- [ClrOnlyFact]: Only run the test on Desktop CLR

These attributes don't reflect a completely well thought out plan and certainly can't hit all of the combinations that I listed above.  It's more of me adding a few things as I was making progress in our getting our code to run tests on Linux / Mac.  

At some point post RTM we're going to be sitting down and re-evaluating these attributes to come up with a more sensible structure.

**Pull request merged**